### PR TITLE
feature(backend): Add support for accelerator config

### PR DESCRIPTION
**Pull request by gkcalat**
**Description of your changes:**

[PipelineSpec has](https://github.com/kubeflow/pipelines/blob/a4abf27df478ca9a970ea4e4083adcd231ddda7e/api/v2alpha1/pipeline_spec.proto#L705-L711) `AcceleratorConfig` that specifies `Type` and `Count` for accelerator resources limits.

This PR adds support in KFP backend for the following vendors:
1. `nvidia.com/gpu`
2. `amd.com/gpu`
3. `cloud-tpus.goocle.com/v2` and `cloud-tpus.goocle.com/v3`
4. Custom vendors specified by users. E.g. `gpu.intel.com/i915`, `fpga.intel.com/v2`, `gpu-vendor.example/example-gpu`, etc.

Besides, we support accelerator type strings that are present in samples. For example, `TPU_V3`, `tpu-v3`, `nvidia-tesla-a100`, `NVIDIA_TESLAT_A100`, etc.

**Checklist:**
- [x] Manually tested on GCP for `nvidia.comn/gpu` devices (k80 and v100)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by chensun — CHANGES_REQUESTED** [review]
Generally speaking we should have test for new code, especially for new features. Secondly, should we do this together with the node selector part? WDYT?

File: backend/src/v2/driver/driver.go [hunk 1]
```diff
@@ -364,8 +364,42 @@ func makePodSpecPatch(
 	}
 	accelerator := container.GetResources().GetAccelerator()
 	if accelerator != nil {
-		return "", fmt.Errorf("accelerator resources are not supported yet: https://github.com/kubeflow/pipelines/issues/7043")
+		var acceleratorType k8score.ResourceName
+		if temp := strings.ToLower(accelerator.GetType()); temp != "" {
+			switch {
+			case strings.Contains(temp, "nvidia"):
```
[the remark this answers is not in the document]
Comment (gkcalat) [reply backend/src/v2/driver/driver.go]: Synced offline. We decided to remove the custom type handling and limit the scope of this PR to `resource.limits`. `nodeSelector`  will need to be general and not limited to the scope of accelerator config.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks for adding the unit test coverage.

**Pull request merged**